keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
-    // particular driver.
-    StructuredData *StructuredResourceHandle
-}
-
-// StructuredResourceHandle is the in-tree representation of the allocation result.
-type StructuredResourceHandle struct {
-    // VendorClassParameters are the per-claim configuration parameters
-    // from the resource class at the time that the claim was allocated.
-    VendorClassParameters runtime.Object
+    // Config contains all the configuration pieces that apply to the request
+    // and that were meant for the driver which handles these devices.
+    // They get collected during the allocation and stored here
+    // to ensure that they remain available while the claim is allocated.
+    //
+    // Entries are list in the same order as in class.config and claim.config,
+    // with class.config entries first.
+    //
+    // +optional
+    Config []DeviceConfiguration
 
-    // VendorClaimParameters are the per-claim configuration parameters
-    // from the resource claim parameters at the time that the claim was
-    // allocated.
-    VendorClaimParameters runtime.Object
+    // RequestName identifies the request in the claim which caused this
+    // device to be allocated. If a request was satisfied by multiple
+    // devices, they are indexed starting with zero in the order
+    // in which they are listed in the results.
+    RequestName string
 
-    // NodeName is the name of the node providing the necessary resources
-    // if the resources are local to a node.
-    NodeName string
+    // This name together with the driver name and the device name field
+    // identify which device was allocated.
+    //
+    // For a driver with node-local devices, it is the name of the node and
+    // the overall ID is `<driver name>/<node name>/<device name>`.
+    //
+    // For other drivers, the overall ID is `<driver name>[/pool
+    // name>]/<device name>`.
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +optional
+    PoolName string
 
-    // Results lists all allocated driver resources.
-    Results []DriverAllocationResult
+    // DeviceName references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    DeviceName string
 }
 
-// DriverAllocationResult contains vendor parameters and the allocation result for
-// one request.
-type DriverAllocationResult struct {
-    // VendorRequestParameters are the per-request configuration parameters
-    // from the time that the claim was allocated.
-    VendorRequestParameters runtime.Object
+const PoolNameMaxLength = validation.DNS1123SubdomainMaxLength // Same as for a single node name.
+
+// DeviceConfiguration is one entry in a list of configuration pieces for a device.
+type DeviceConfiguration struct {
+    // Admins is true if the source of the piece was a class and thus
+    // not something that a normal user would have been able to set.
```

Comment:
source of the piece?

Reply:
If this means "came from the class" maybe we should just name it that way `ClassConfig` or something?

Reply:
"configuration piece", or just "configuration".

Reply:
Let's drop the "piece".

Reply:
I'm using "configuration parameters" when I need a plural form. "Configurations" looked odd.

Reply:
I've not renamed it to `ClassConfig` because I prefer to describe the semantic instead of the mechanism.